Initial TDD Finish

zyelcrier opened the pull request:
- three helper functions to make the main function more readable
-first test for edge cases, then runs program

zyelcrier committed:
Initial TDD Finish

zyelcrier commented:
Potential idea is to take out initial if else because those cases are covered in the helper functions all tests pass without them. Thoughts? (Initial if else in reference to lines 33 and 66)

zyelcrier committed:
Remove Unecisarry code

rayliu0317 commented on walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:
```diff
@@ -15,9 +15,84 @@
+    Collection<Event> eventsWithRequestedAttendees = getEventsWithRequestedAttendees(events, request.getAttendees());
+    Collection<TimeRange> availabeRanges = new ArrayList<>();
+
+    if(events.size()==0 && meetingRequestDuration<=TimeRange.WHOLE_DAY.end()){ 
```
There should be spaces between if and (. And it should be applied to the rest of the file also.

zyelcrier replied:
I believed i fixed the space you were referring to. Sure we can go through that as well in our 1:1 but do you think that enough time to explain complexity. I also have to do the second part of the week 5 project. @rayliu0317

rayliu0317 reviewed (commented):
I think this works and is fine for an initial attempt. We can go over the optimal algorithm in our 1:1 on Monday if you want.

zyelcrier committed:
Add Space

zyelcrier committed:
Finish Tests

zyelcrier committed:
Review Changes

zyelcrier commented:
Review Changes:
-Added proper descriptions to the five tests I made 
-changed variable name on collection of time ranges to be more specific
@rayliu0317

rayliu0317 commented on walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:
```diff
@@ -30,11 +30,11 @@
-    Collection<TimeRange> availabeRangesForAllAttendees = findAllAvailableTimeRanges(eventsWithAllAttendees);
-    Collection<TimeRange> result02 = findViableTimeRangesForRequest(availabeRangesForAllAttendees, meetingRequestDuration);
+    Collection<TimeRange> availabeRangesForRequestedAndOptionalAttendees = findAllAvailableTimeRanges(eventsWithAllAttendees);
+    Collection<TimeRange> result02 = findViableTimeRangesForRequest(availabeRangesForRequestedAndOptionalAttendees, meetingRequestDuration);
```
Still not clear to me what result01 and result02 are supposed to represent.

zyelcrier replied:
I changed those to be specific  I didnt see anything else that I thought was vague or un-descriptive. Does the new change work ? @rayliu0317

zyelcrier committed:
Variable Name Change
